### PR TITLE
Update WC property suppressHtmlRecursiveErrorOutput

### LINKED ISSUE #15726: Re-introduce change reverted from 14248
Re-introduce change reverted from https://github.com/OpenLiberty/open-liberty/pull/14248

### PR DESCRIPTION
**Pull request by pmd1nh**
fixes #15726

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build        #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -63,9 +63,6 @@ exposeWebInfOnDispatch.desc=If true, a servlet can access files in the WEB-INF d
 decodeUrlPlusSign.name=Decode url plus sign
 decodeUrlPlusSign.desc=Decode the plus sign when it is part of the URL. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.decodeurlplussign. For servlet-5.0 and newer Servlet features, the default value is false.
 
-suppressHtmlRecursiveErrorOutput.name=Suppress recursive exception in the HTML response
-suppressHtmlRecursiveErrorOutput.desc=Suppresses the exception information from appearing in the HTML output when there is a recursive error that cannot be handled by an application's configured error page. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.suppressHtmlRecursiveErrorOutput.
-
 fileWrapperEvents.name=File wrapper events
 fileWrapperEvents.desc=Web container will generate SMF and PMI data when  serving the static files. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.fileWrapperEvents.
```
Comment (dmuelle) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:67]: suggested edit ( if possible)

The web container generates SMF and PMI data when it serves the static files.

**Comment by andrea-mh** [comment]
No message review requested from L2

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
Note that delete metatype.xml message does not need translation

**Comment by pmd1nh** [comment]
Procedure when remove an existence metatype.xml property/entry:

1. Update the CL MetaTypeValidatorTest
(see example: https://github.ibm.com/websphere/WS-CD-Open/pull/24091)

integrate and wait til it is in a CL build

2. Submit PB for OL side and make sure there is no **ant_schemaValidator unittest failure**
3. Integrate OL.

**Comment by pmd1nh** [comment]
Change in CL to suppress the validator test for the removal of suppressHtmlRecursiveErrorOutput in metatype.xml

https://github.ibm.com/websphere/WS-CD-Open/pull/24091

**Comment by pmd1nh** [comment]
https://github.ibm.com/websphere/WS-CD-Open/pull/24091 is merged.  Try build and fat again

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
ant_schemaValidator unittests passed.
All WC tests passed

On failure but does not look connect to this change.
wlp.lib.extract_fat FAT tests | 15 | 1 | 0 | &nbsp; | &nbsp;
-- | -- | -- | -- | -- | --

File: dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java [hunk 2]
```diff
@@ -70,7 +70,7 @@
 
     public static String ERROR_PAGE_COMPATIBILITY;
     public static boolean MAP_FILTERS_TO_ASTERICK;
-    public static boolean SUPPRESS_HTML_RECURSIVE_ERROR_OUTPUT;
+    public static boolean SUPPRESS_HTML_RECURSIVE_ERROR_OUTPUT = true;         //Property is not read by the server starting 21.0.0.6; default is always true
     public static boolean THROW_MISSING_JSP_EXCEPTION;          //PK57843
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java:74]: This needs to be updated to the correct release 21.0.0.7 for now.

File: dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java [hunk 3]
```diff
@@ -542,7 +542,7 @@ public static void setCustomPropertyVariables() {
 
         ERROR_PAGE_COMPATIBILITY = customProps.getProperty("com.ibm.ws.webcontainer.contenttypecompatibility");
         MAP_FILTERS_TO_ASTERICK = Boolean.valueOf(customProps.getProperty("com.ibm.ws.webcontainer.mapfilterstoasterisk")).booleanValue();
-        SUPPRESS_HTML_RECURSIVE_ERROR_OUTPUT = Boolean.valueOf(customProps.getProperty("com.ibm.ws.webcontainer.suppresshtmlrecursiveerroroutput")).booleanValue();
+        //SUPPRESS_HTML_RECURSIVE_ERROR_OUTPUT = Boolean.valueOf(customProps.getProperty("com.ibm.ws.webcontainer.suppresshtmlrecursiveerroroutput")).booleanValue(); // issue 14242, no longer read by the server started 21.0.0.6
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java]: Also needs an updated release version.

**Review by pnicolucci — COMMENTED** [review]
Just the two minor changes, otherwise looks good.

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
#run-libby-bot

**Comment by pmd1nh** [comment]
these failure do not look to connect with this change.

com.ibm.ws.microprofile.rest.client_fat FAT tests | 75 | 1 | 0 | &nbsp; | &nbsp;
-- | -- | -- | -- | -- | --
com.ibm.ws.security.wim.adapter.ldap_fat FAT tests | 1 | 0 | 1 | &nbsp; | &nbsp;
com.ibm.ws.security.wim.adapter.ldap_fat javacores | 1 | 1 | 0 | &nbsp; | &nbsp;
wlp.lib.extract_fat FAT tests | 15 | 1 | 0 | &nbsp; | &nbsp;

**Review by pnicolucci — APPROVED** [review]
Changes look good. Just ensure to review the personal build to ensure no failures are related to these changes. Seems unlikely given the changes  and the failures.

**Comment by pmd1nh** [comment]
com.ibm.ws.fat.wc.tests.WC5JakartaServletTest | 3 | 0 | 0 | 2 s | 19
-- | -- | -- | -- | -- | --
test_Simple_Servlet50_EE9_FEATURES | 1 | 0 | 0 | 1 s | 1
test_SuppressHtmlRecursiveErrorOutput_EE9_FEATURES | 1 | 0 | 0 | 0 s | 2
&nbsp;

**Pull request merged**